NanaZip.UI.Classic/SevenZip/CPP/7zip/UI/FileManager/PanelFolderChange.cpp

```diff
@@ -72,10 +72,48 @@ static bool DoesNameContainWildcard_SkipRoot(const UString &path)
 {
   return DoesNameContainWildcard(path.Ptr(NName::GetRootPrefixSize(path)));
 }
+bool HasDriveLetter(const UString& path)
+{
+    return path.Len() >= 2 &&
+        ((path[0] >= L'A' && path[0] <= L'Z') ||
+         (path[0] >= L'a' && path[0] <= L'z')) &&
+        path[1] == L':';
+}
+
+UString ExpandFirstEnvInPath(const UString &path)
+{
+    int firstPercent = path.Find(L'%');
+    if (firstPercent == std::wstring::npos)
+        return path;
+    int secondPercent = path.Find(L'%', firstPercent + 1);
+    if (secondPercent == std::wstring::npos)
+        return path;
+    UString envName;
+    for (int i = firstPercent; i <= secondPercent; ++i)
+        envName += path[i];
+    wchar_t buffer[MAX_PATH * 4] = {};
+    DWORD ret = ExpandEnvironmentStringsW(envName.Ptr(),
+     buffer, ARRAYSIZE(buffer));
+    UString expanded = (ret == 0 || ret > ARRAYSIZE(buffer))?
+                            envName : buffer;
+
+    UString result;
+    if (firstPercent > 0)
+        result += path.Left(firstPercent);
+    result += expanded;
+    if (secondPercent + 1 <(int) path.Len())
+        result += path.Ptr(secondPercent + 1);
+    return result;
+}
 
-HRESULT CPanel::BindToPath(const UString &fullPath, const UString &arcFormat, COpenResult &openRes)
+HRESULT CPanel::BindToPath(const UString &fullPath, 
+  const UString &arcFormat, COpenResult &openRes)
 {
-  UString path = fullPath;
+  UString path ;
```

Comment:
Use "::" prefix for functions in the global namespace to avoid name conflicts with other namespaces if you are using C++.

There are similar issues.